src/enums.rs

```diff
@@ -42,7 +44,7 @@ pub fn int_to_event_type(code: u32) -> Option<EventType> {
 
 #[allow(non_camel_case_types)]
 #[cfg_attr(feature = "serde", derive(Serialize), derive(Deserialize))]
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
 pub enum EventCode {
```

Comment:
All this should be Copy, Eq and Hash. Copy because it is small (12 bytes on x64, smaller than i128, which is also Copy). Eq and Hash seemed like a good idea, but I might be missing something.